frontend/src/v5/ui/components/shared/userMenu/editProfileModal/editProfilePersonalTab/editProfilePersonalTab.component.tsx

```diff
@@ -0,0 +1,217 @@
+import { ScrollAreaPadding } from './editProfilePersonal.styles';
+
+interface IUpdatePersonalInputs {
+	firstName: string;
+	lastName: string;
+	email: string;
+	company: string;
+	countryCode: string;
+}
+
+type EditProfilePersonalTabProps = {
+	setSubmitFunction: (fn: Function) => void,
+	setIsSubmitting: (isSubmitting: boolean) => void,
+	user: IUser,
+};
+
+export const EditProfilePersonalTab = ({
+	setSubmitFunction,
+	setIsSubmitting,
+	user,
+}: EditProfilePersonalTabProps) => {
+	const personalError = CurrentUserHooksSelectors.selectPersonalError();
+	const formIsUploading = CurrentUserHooksSelectors.selectPersonalDataIsUpdating();
+	const [newAvatarFile, setNewAvatarFile] = useState(null);
+	const [alreadyExistingEmails, setAlreadyExistingEmails] = useState([]);
+	const [unexpectedError, setUnexpectedError] = useState(false);
+
+	const trimPersonalValues = (personalValues: IUpdatePersonalInputs): IUpdatePersonalInputs => transform(
+		personalValues,
+		// eslint-disable-next-line no-param-reassign
+		(result, value, key) => { result[key] = value.trim(); },
+		{} as IUpdatePersonalInputs,
+	);
+
+	const getUserPersonalValues = () => trimPersonalValues(
+		pick(
+			defaults(user, { company: '', countryCode: 'GB' }),
+			['firstName', 'lastName', 'email', 'company', 'countryCode'],
+		),
+	);
+
+	const {
+		getValues,
+		trigger,
+		handleSubmit,
+		reset,
+		watch,
+		control,
+		formState: { errors, isValid: formIsValid, isSubmitted, isSubmitSuccessful },
+	} = useForm<IUpdatePersonalInputs>({
+		mode: 'all',
+		resolver: yupResolver(EditProfileUpdatePersonalSchema(alreadyExistingEmails)),
+		defaultValues: getUserPersonalValues(),
+	});
+
+	const firstName = watch('firstName');
+	const lastName = watch('lastName');
+	const company = watch('company');
+
+	const getTrimmedValues = () => trimPersonalValues(getValues());
+
+	const onSubmit = () => {
+		setUnexpectedError(false);
+		const values = getTrimmedValues();
+		if (!values.company) {
+			delete values.company;
+		}
+		CurrentUserActionsDispatchers.updatePersonalData({
+			...values,
+			avatarFile: newAvatarFile,
+		});
+	};
+
+	const uploadWasSuccessful = !formIsUploading && !personalError;
+
+	const fieldsAreDirty = () => !isMatch(user, getTrimmedValues()) || newAvatarFile;
+
+	// enable submission only if form is valid and fields are dirty (or avatar was changed)
+	useEffect(() => {
+		const shouldEnableSubmit = formIsValid && fieldsAreDirty();
+		setSubmitFunction(() => (shouldEnableSubmit ? handleSubmit(onSubmit) : null));
+	}, [newAvatarFile, firstName, lastName, company]);
+
+	// update form values when user is updated
+	useEffect(() => {
+		if (uploadWasSuccessful && isSubmitSuccessful) {
+			if (newAvatarFile) {
+				setNewAvatarFile(null);
+			}
+			reset(getUserPersonalValues(), { keepIsSubmitted: true });
+		}
+	}, [formIsUploading]);
+
+	useEffect(() => {
+		if (personalError) {
+			if (personalError?.message === 'Email already exists') {
```

Comment:
What is the reason for adding the email to `alreadyExistingEmails` here?

Reply:
If a user tries to update their current password, but the new one is already in use a message needs to be printed. To improve the UX, the tried password are temporarily stored so that the user is prevented from trying the same email twice